### ISSUE TITLE
[BUG]Unexpected results if negative axis in mt.linalg.norm

### ISSUE THREAD

**Issue by YoshieraHuang** [mars-project/mars#1555]
## Describe the bug
The result of `mt.linalg norm` is different from `np.linalg.norm` if axis is negative. See below for details:
```
In [21]: np.linalg.norm(np.random.rand(1330,3), axis=-1)                                                                                                                                             
Out[21]: 
array([0.8566685 , 1.31623791, 0.66768178, ..., 0.50399358, 0.51119494,
       0.74094714])

In [22]: np.linalg.norm(np.random.rand(1330,3), axis=-1).shape                                                                                                                                       
Out[22]: (1330,)

In [24]: mt.linalg.norm(mt.random.rand(1330,3), axis=-1).shape                                                                                                                                       
Out[24]: (1330, 3)

In [25]: mt.linalg.norm(mt.random.rand(1330,3), axis=-1).execute()                                                                                                                                   
Out[25]: ---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~/anaconda3/lib/python3.7/site-packages/IPython/core/formatters.py in __call__(self, obj)
    700                 type_pprinters=self.type_printers,
    701                 deferred_pprinters=self.deferred_printers)
--> 702             printer.pretty(obj)
    703             printer.flush()
    704             return stream.getvalue()

~/anaconda3/lib/python3.7/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    392                         if cls is not object \
    393                                 and callable(cls.__dict__.get('__repr__')):
--> 394                             return _repr_pprint(obj, self, cycle)
    395 
    396             return _default_pprint(obj, self, cycle)

~/anaconda3/lib/python3.7/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    698     """A pprint that just redirects to the normal repr function."""
    699     # Find newlines and replace them with p.break_()
--> 700     output = repr(obj)
    701     lines = output.splitlines()
    702     with p.group():

~/anaconda3/lib/python3.7/site-packages/mars/core.py in __repr__(self)
    126 
    127     def __repr__(self):
--> 128         return self._data.__repr__()
    129 
    130     def _check_data(self, data):

~/anaconda3/lib/python3.7/site-packages/mars/tensor/core.py in __repr__(self)
    160 
    161     def __repr__(self):
--> 162         return self._to_str(representation=True)
    163 
    164     @property

~/anaconda3/lib/python3.7/site-packages/mars/tensor/core.py in _to_str(self, representation)
    148             threshold = print_options['threshold']
    149 
--> 150             corner_data = fetch_corner_data(self, session=self._executed_sessions[-1])
    151             # if less than default threshold, just set it as default,
    152             # if not, set to corner_data.size - 1 make sure ... exists in repr

~/anaconda3/lib/python3.7/site-packages/mars/tensor/utils.py in fetch_corner_data(tensor, session)
    817             corners[indices] = tensor[tuple(slc)]
    818         # fetch together
--> 819         fetched = ExecutableTuple(corners.flat).fetch(session=session)
    820         for indices, f in zip(indices_iter, fetched):
    821             corners[indices] = f

~/anaconda3/lib/python3.7/site-packages/mars/core.py in fetch(self, session, **kw)
    675         if len(self) == 0:
    676             return tuple()
--> 677         return super().fetch(session=session, **kw)
    678 
    679     def _attach_session(self, session):

~/anaconda3/lib/python3.7/site-packages/mars/core.py in fetch(self, session, **kw)
    377         if session is None:
    378             session = Session.default_or_local()
--> 379         return session.fetch(self, **kw)
    380 
    381     def _attach_session(self, session):

~/anaconda3/lib/python3.7/site-packages/mars/session.py in fetch(self, *tileables, **kw)
    489             ret_list = True
    490 
--> 491         result = self._sess.fetch(*tileables, **kw)
    492 
    493         ret = []

~/anaconda3/lib/python3.7/site-packages/mars/session.py in fetch(self, n_parallel, *tileables, **kw)
    122         if n_parallel is None:
    123             kw['n_parallel'] = cpu_count()
--> 124         return self._executor.fetch_tileables(tileables, **kw)
    125 
    126     def create_mutable_tensor(self, name, shape, dtype, fill_value=None, *args, **kwargs):

~/anaconda3/lib/python3.7/site-packages/mars/utils.py in _wrapped(*args, **kwargs)
    406                 _kernel_mode.eager = False
    407             _kernel_mode.eager_count = enter_eager_count + 1
--> 408             return func(*args, **kwargs)
    409         finally:
    410             _kernel_mode.eager_count -= 1

~/anaconda3/lib/python3.7/site-packages/mars/executor.py in fetch_tileables(self, tileables, **kw)
    970 
    971         # if chunk executed, fetch chunk mechanism will be triggered in execute_tileables
--> 972         result = self.execute_tileables(tileables, **kw)
    973         for to_release_tileable in to_release_tileables:
    974             for c in get_tiled(to_release_tileable, mapping=tileable_optimized).chunks:

~/anaconda3/lib/python3.7/site-packages/mars/utils.py in _wrapped(*args, **kwargs)
    406                 _kernel_mode.eager = False
    407             _kernel_mode.eager_count = enter_eager_count + 1
--> 408             return func(*args, **kwargs)
    409         finally:
    410             _kernel_mode.eager_count -= 1

~/anaconda3/lib/python3.7/site-packages/mars/utils.py in inner(*args, **kwargs)
    500     def inner(*args, **kwargs):
    501         with build_mode():
--> 502             return func(*args, **kwargs)
    503     return inner
    504 

~/anaconda3/lib/python3.7/site-packages/mars/executor.py in execute_tileables(self, tileables, fetch, n_parallel, n_thread, print_progress, mock, compose, name)
    878                                    n_parallel=n_parallel or n_thread,
    879                                    print_progress=print_progress, mock=mock,
--> 880                                    chunk_result=chunk_result)
    881 
    882                 # update shape of tileable and its chunks whatever it's successful or not

~/anaconda3/lib/python3.7/site-packages/mars/executor.py in execute_graph(self, graph, keys, n_parallel, print_progress, mock, no_intermediate, compose, retval, chunk_result)
    688             print_progress=print_progress, mock=mock, mock_max_memory=self._mock_max_memory,
    689             fetch_keys=fetch_keys, no_intermediate=no_intermediate)
--> 690         res = graph_execution.execute(retval)
    691         self._mock_max_memory = max(self._mock_max_memory, graph_execution._mock_max_memory)
    692         if mock:

~/anaconda3/lib/python3.7/site-packages/mars/executor.py in execute(self, retval)
    569         # wait until all the futures completed
    570         for future in executed_futures:
--> 571             future.result()
    572 
    573         if retval:

~/anaconda3/lib/python3.7/concurrent/futures/_base.py in result(self, timeout)
    426                 raise CancelledError()
    427             elif self._state == FINISHED:
--> 428                 return self.__get_result()
    429 
    430             self._condition.wait(timeout)

~/anaconda3/lib/python3.7/concurrent/futures/_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

~/anaconda3/lib/python3.7/concurrent/futures/thread.py in run(self)
     55 
     56         try:
---> 57             result = self.fn(*self.args, **self.kwargs)
     58         except BaseException as exc:
     59             self.future.set_exception(exc)

~/anaconda3/lib/python3.7/site-packages/mars/executor.py in _execute_operand(self, op)
    441             # so we pass the first operand's first output to Executor.handle
    442             first_op = ops[0]
--> 443             Executor.handle(first_op, results, self._mock)
    444 
    445             # update maximal memory usage during execution

~/anaconda3/lib/python3.7/site-packages/mars/executor.py in handle(cls, op, results, mock)
    639                 # The `UFuncTypeError` was introduced by numpy#12593 since v1.17.0.
    640                 try:
--> 641                     return runner(results, op)
    642                 except UFuncTypeError as e:
    643                     raise TypeError(str(e)).with_traceback(sys.exc_info()[2]) from None

~/anaconda3/lib/python3.7/site-packages/mars/tensor/indexing/getitem.py in execute(cls, ctx, op)
     83         input_ = ctx[op.inputs[0].key]
     84         xp = get_array_module(input_)
---> 85         ret = xp.asarray(input_)[indexes]
     86         if hasattr(ret, 'astype'):
     87             ret = ret.astype(

IndexError: too many indices for array
```

**Comment by qinxuye** [comment]
Failed when displaying data, the execution should have finished. The possible issue is that the shape of `norm` is wrong.

For display purpose, Mars only fetch the corner data only, if the shape is wrong, the failure may emerge.

**Comment by YoshieraHuang** [comment]
> Failed when displaying data, the execution should have finished. The possible issue is that the shape of `norm` is wrong.
> 
> For display purpose, Mars only fetch the corner data only, if the shape is wrong, the failure may emerge.

Exactly, the wrong shape is the matter. Could you fix this please?

**Comment by qinxuye** [comment]
> > Failed when displaying data, the execution should have finished. The possible issue is that the shape of `norm` is wrong.
> > For display purpose, Mars only fetch the corner data only, if the shape is wrong, the failure may emerge.
> 
> Exactly, the wrong shape is the matter. Could you fix this please?

OK, I am working on it.

**Comment by qinxuye** [comment]
@YoshieraHuang I found that the master branch is fine, and I have checked that it's been resolved by #1499, maybe you didn't rebase the code?

Could you please check that again?

**Comment by YoshieraHuang** [comment]
> @YoshieraHuang I found that the master branch is fine, and I have checked that it's been resolved by #1499, maybe you didn't rebase the code?
> 
> Could you please check that again?

It works fine in the master branch. I will closs this issue. Thank you very much!!
